core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java

```diff
@@ -970,6 +976,51 @@ public void testCatalogWithCustomMetricsReporter() throws IOException {
     Assertions.assertThat(CustomMetricsReporter.COUNTER.get()).isEqualTo(2);
   }
 
+  @Test
+  public void testCommitExceptionWithoutMessage() {
```

Comment:
can be simplified to 
```
@Test
  public void testCommitExceptionWithoutMessage() {
    TableIdentifier tableIdent = TableIdentifier.of("db", "tbl");
    BaseTable table = (BaseTable) catalog.buildTable(tableIdent, SCHEMA).create();
    TableOperations ops = table.operations();
    TableMetadata metadataV1 = ops.current();

    table.updateSchema().addColumn("n", Types.IntegerType.get()).commit();
    ops.refresh();

    try (MockedStatic<JdbcUtil> mockedStatic = Mockito.mockStatic(JdbcUtil.class)) {
      mockedStatic
          .when(() -> JdbcUtil.loadTable(any(), any(), any(), any()))
          .thenThrow(new SQLException());
      assertThatThrownBy(() -> ops.commit(ops.current(), metadataV1))
          .isInstanceOf(UncheckedSQLException.class)
          .hasMessageStartingWith("Unknown failure");
    }
  }
```

please also update the other tests accordingly. The spying on `ops` isn't necessary here, because you're throwing the exception when the table/view is being loaded (which is different from https://github.com/apache/iceberg/pull/10069)